request.go

```diff
@@ -35,6 +36,15 @@ func New(address string) *Req {
 		Method: "GET",
 		Header: make(http.Header),
 	}
+	if ctx != nil {
```

Comment:
Well the following usage is a better practice. Lets change it this way:

```
if ctx == nil {
	ctx = context.Background()
}

r.request = (&http.Request{
	Method: "GET",
	Header: make(http.Header),
}).WithContext(ctx)
```

Reply:
Fixed in https://github.com/binalyze/httpreq/pull/12/commits/afbc08a0ae51ce5a56639f7906ecfc65e0f18886